### PR TITLE
Fix xref2 tests

### PR DESCRIPTION
**Pull request by Julow**
I also added the tests to `make test`.
Should we move these tests to the root `test` directory ?

### CODE REVIEW

File: src/xref2/scratch.md [hunk 1]
```diff
@@ -125,7 +127,7 @@ let sg = Common.model_of_string_impl test_data;;
                     `Module
                       (`Result
                          (`Result (`Module (`Root (Common.root, "Root"), M))),
-                       E/hidden);
+                       $E);
```
Comment (Julow) [inline src/xref2/scratch.md:130]: Is this change caused by the version of OCaml ? I'm using 4.08.1.

File: src/xref2/test/lib/common.cppo.ml [hunk 2]
```diff
@@ -45,10 +45,24 @@ let root =
...
+let root_identifier = `Identifier root_with_name
+
+let root_module name = `Module (root_with_name, Odoc_model.Names.ModuleName.of_string name)
 
 let root_pp fmt (_ : Odoc_model.Root.t) = Format.fprintf fmt "Common.root"
 
+let _quoted fmt_a fmt a = Format.fprintf fmt "\"%a\"" fmt_a a
+let _quoted_str a_to_str fmt a = Format.fprintf fmt "%S" (a_to_str a)
+
+let unit_name_pp = _quoted_str Odoc_model.Names.UnitName.to_string
```
Comment (jonludlam) [inline src/xref2/test/lib/common.cppo.ml:59]: There are formatters for these - `Odoc_model.Names.UnitName.fmt` - we could just use those (which would drop the quote marks though)

Comment (Julow) [reply src/xref2/test/lib/common.cppo.ml:59]: Some of these are of module type `SimpleName` which doesn't have that function.

File: src/xref2/scratch.md (hunk 1 above)
Comment (jonludlam) [reply src/xref2/scratch.md:130]: No, this is due to the rebase - I've changed from using `/hidden` to using the `InternalName` things.

File: src/xref2/test/lib/common.cppo.ml (hunk 2 above)
Comment (jonludlam) [reply src/xref2/test/lib/common.cppo.ml:59]: We should probably add `fmt` to `SimpleName`

Comment (Julow) [reply src/xref2/test/lib/common.cppo.ml]: I added `fmt` and used it directly, dropping the quotes. The diffs are bigger but that's not important I think

**Pull request merged**